drivers/sensor/adxl345/adxl345.c

```diff
@@ -116,40 +125,37 @@ static int adxl345_read_sample(const struct device *dev,
 		return rc;
 	}
 
-	raw_x = axis_data[0] | (axis_data[1] << 8);
-	raw_y = axis_data[2] | (axis_data[3] << 8);
-	raw_z = axis_data[4] | (axis_data[5] << 8);
-
-	sample->x = raw_x;
-	sample->y = raw_y;
-	sample->z = raw_z;
+	sample->x = axis_data[0] | (((int16_t)axis_data[1]) << 8);
+	sample->y = axis_data[2] | (((int16_t)axis_data[3]) << 8);
+	sample->z = axis_data[4] | (((int16_t)axis_data[5]) << 8);
 
 	return 0;
 }
 
-static void adxl345_accel_convert(struct sensor_value *val, int16_t sample)
+static inline void adxl345_accel_convert(struct sensor_value *val, int16_t sample, uint16_t scale)
 {
-	if (sample & BIT(9)) {
-		sample |= ADXL345_COMPLEMENT;
-	}
+	int64_t micro_g = (sample * scale) * 1000;
 
-	val->val1 = ((sample * SENSOR_G) / 32) / 1000000;
-	val->val2 = ((sample * SENSOR_G) / 32) % 1000000;
+	sensor_ug_to_ms2(micro_g, val);
 }
 
-static int adxl345_sample_fetch(const struct device *dev,
-				enum sensor_channel chan)
+static int adxl345_sample_fetch(const struct device *dev, enum sensor_channel chan)
 {
 	struct adxl345_dev_data *data = dev->data;
 	struct adxl345_sample sample;
 	uint8_t samples_count;
 	int rc;
 
 	data->sample_number = 0;
-	rc = adxl345_reg_read_byte(dev, ADXL345_FIFO_STATUS_REG, &samples_count);
-	if (rc < 0) {
-		LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
-		return rc;
+
+	if (data->fifo_mode == ADXL345_FIFO_BYPASS) {
+		samples_count = 1;
+	} else {
+		rc = adxl345_reg_read_byte(dev, ADXL345_FIFO_STATUS_REG, &samples_count);
+		if (rc < 0) {
+			LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+			return rc;
+		}
```

Comment:
Also not related to devicetree. Please revert

Reply:
This change is needed to fetch samples properly if FIFO is disabled.

Because if FIFO is disabled, reading from ADXL345_FIFO_STATUS_REG returns 0.
If sample_count is zero, the driver will never fetch a sample within the loop body.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/16085873/ba5fd589-3272-4542-938a-3ddb5b4a022e)

Reply:
This behavior will be fixed with #67499